Index: u-boot-2025.01/arch/arm/dts/sun8i-t113s-nc-nagami.dts
===================================================================
--- /dev/null
+++ u-boot-2025.01/arch/arm/dts/sun8i-t113s-nc-nagami.dts
@@ -0,0 +1,29 @@
+/dts-v1/;
+#include "sun8i-t113s.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+/ {
+    model = "Tronlong TLT113-MiniEVM (MMC-Fixed)";
+    compatible = "tronlong,tlt113-minievm", "allwinner,sun8i-t113i", "allwinner,sun8i-t113s";
+    aliases { serial0 = &uart0; mmc0 = &mmc0; };
+    chosen { stdout-path = "serial0:115200n8"; };
+    reg_vcc3v3: vcc3v3 {
+        compatible = "regulator-fixed";
+        regulator-name = "vcc-3v3";
+        regulator-min-microvolt = <3300000>;
+        regulator-max-microvolt = <3300000>;
+        regulator-always-on;
+        bootph-all;
+    };
+};
+&ccu { bootph-all; };
+&pio { bootph-all; };
+&uart0 { pinctrl-names = "default"; pinctrl-0 = <&uart0_pg_pins>; status = "okay"; bootph-all; };
+&mmc0 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&mmc0_pins>;
+    vmmc-supply = <&reg_vcc3v3>;
+    bus-width = <4>;
+    broken-cd;
+    status = "okay";
+    bootph-all;
+};
