Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -459,6 +459,12 @@ u32 spl_mmc_boot_mode(struct mmc *mmc, c
 
 void board_init_f(ulong dummy)
 {
+	{ /* Step 1: Violent PC0 LED On */
+		*(volatile unsigned int *)0x0200190C = 0x00010001; /* 1. PIO Clock Gate */
+		*(volatile unsigned int *)0x02000128 = 0x00000070; /* 2. Port G/C/F Power Gate */
+		*(volatile unsigned int *)0x02000060 = (*(volatile unsigned int *)0x02000060 & ~0xF) | 1; /* 3. PC0 CFG -> Output */
+		*(volatile unsigned int *)0x02000070 |= 1; /* 4. PC0 DATA -> High */
+	}
 	sunxi_sram_init();
 
 	/* Enable non-secure access to some peripherals */
