Index: u-boot-2025.01/arch/arm/dts/sun8i-t113-tronlong.dts
===================================================================
--- /dev/null
+++ u-boot-2025.01/arch/arm/dts/sun8i-t113-tronlong.dts
@@ -0,0 +1,34 @@
+/dts-v1/;
+#include "sun8i-t113s.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+/ {
+    model = "Tronlong TLT113-MiniEVM";
+    compatible = "tronlong,tlt113-minievm", "allwinner,sun8i-t113i", "allwinner,sun8i-t113s";
+    aliases { 
+        serial0 = &uart3; // 关键：将 serial0 映射到 uart3
+        mmc0 = &mmc0; 
+    };
+    chosen { stdout-path = "serial0:115200n8"; };
+};
+&ccu { bootph-all; };
+
+/* 配置 PG17/PG18 为 UART3 (Function 7) */
+&pio {
+    bootph-all;
+    uart3_pg_pins: uart3-pg-pins { 
+        pins = "PG17", "PG18"; 
+        function = "uart3"; 
+        bootph-all; 
+    };
+    mmc0_pins: mmc0-pins { pins = "PF0", "PF1", "PF2", "PF3", "PF4", "PF5"; function = "sdc0"; drive-strength = <30>; bias-pull-up; bootph-all; };
+};
+
+/* 启用 UART3 */
+&uart3 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&uart3_pg_pins>;
+    status = "okay";
+    bootph-all;
+};
+
+&mmc0 { pinctrl-names = "default"; pinctrl-0 = <&mmc0_pins>; bus-width = <4>; broken-cd; status = "okay"; bootph-all; };
