Index: u-boot-2025.01/arch/arm/dts/sun8i-t113-tronlong.dts
===================================================================
--- /dev/null
+++ u-boot-2025.01/arch/arm/dts/sun8i-t113-tronlong.dts
@@ -0,0 +1,49 @@
+/dts-v1/;
+#include "sun8i-t113s.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+    model = "Tronlong TLT113-MiniEVM";
+    compatible = "tronlong,tlt113-minievm", "allwinner,sun8i-t113i", "allwinner,sun8i-t113s";
+
+    aliases { 
+        serial0 = &uart0; 
+        mmc0 = &mmc0; 
+    };
+    chosen { stdout-path = "serial0:115200n8"; };
+    
+    reg_vcc3v3: vcc3v3 {
+        compatible = "regulator-fixed";
+        regulator-name = "vcc-3v3";
+        regulator-min-microvolt = <3300000>;
+        regulator-max-microvolt = <3300000>;
+        regulator-always-on;
+        bootph-all;
+    };
+};
+
+/* 抄 Nagami: 必须给关键控制器增加 bootph-all，否则 SPL 驱动不加载 */
+&ccu { bootph-all; };
+&pio { bootph-all; };
+
+&uart0 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&uart0_pg_pins>;
+    status = "okay";
+    bootph-all;
+};
+
+&uart0_pg_pins { bootph-all; };
+
+&mmc0 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&mmc0_pins>;
+    vmmc-supply = <&reg_vcc3v3>;
+    bus-width = <4>;
+    broken-cd;       /* 抄 Nagami: 忽略物理 CD 引脚，防止检测失败 */
+    disable-wp;
+    status = "okay";
+    bootph-all;
+};
+
+&mmc0_pins { bootph-all; };
