Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -459,6 +459,40 @@ u32 spl_mmc_boot_mode(struct mmc *mmc, c
 
 void board_init_f(ulong dummy)
 {
+	/* --- BLINK 3 TIMES TEST --- */
+	{
+		volatile unsigned int *ccu_pio = (volatile unsigned int *)0x0200190C;
+		volatile unsigned int *pc_cfg  = (volatile unsigned int *)0x02000060;
+		volatile unsigned int *pc_dat  = (volatile unsigned int *)0x02000070;
+		
+		/* Init LED */
+		*ccu_pio = 0x00010001;
+		*pc_cfg = (*pc_cfg & ~0xF) | 0x1;
+		
+		/* Blink Loop: 3 times */
+		for (int k=0; k<3; k++) {
+			/* ON */
+			*pc_dat |= 0x1;
+			for(volatile int i=0; i<50000000; i++);
+			
+			/* OFF */
+			*pc_dat &= ~0x1;
+			for(volatile int i=0; i<50000000; i++);
+		}
+		
+		/* Try Init DM */
+		if (spl_init() == 0) {
+			/* DM Success: Turn ON forever */
+			struct gpio_desc desc;
+			ofnode node = ofnode_path("/leds/led-0");
+			if (ofnode_valid(node)) {
+				if (!gpio_request_by_name_nodev(node, "gpios", 0, &desc, GPIOD_IS_OUT | GPIOD_IS_OUT_ACTIVE)) {
+					dm_gpio_set_value(&desc, 1);
+				}
+			}
+		}
+		while(1);
+	}
 	sunxi_sram_init();
 
 	/* Enable non-secure access to some peripherals */
