Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -459,6 +459,18 @@ u32 spl_mmc_boot_mode(struct mmc *mmc, c
 
 void board_init_f(ulong dummy)
 {
+	/* --- DTS DM LED TEST --- */
+	{
+		spl_init();
+		struct gpio_desc desc;
+		ofnode node = ofnode_path("/leds/led-0");
+		if (ofnode_valid(node)) {
+			if (!gpio_request_by_name_nodev(node, "gpios", 0, &desc, GPIOD_IS_OUT | GPIOD_IS_OUT_ACTIVE)) {
+				dm_gpio_set_value(&desc, 1);
+			}
+		}
+		while(1);
+	}
 	sunxi_sram_init();
 
 	/* Enable non-secure access to some peripherals */
