Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -469,6 +469,12 @@ void board_init_f(ulong dummy)
 	gpio_init();
 
 	spl_init();
+	{ /* T113-i PG Port Mystery Reg & PIO Gate */
+		*(volatile unsigned int *)0x0200190C = 0x00010001;
+		*(volatile unsigned int *)0x02000128 = (*(volatile unsigned int *)0x02000128 & ~(0xFF << 4)) | (0x77 << 4);
+		*(volatile unsigned int *)0x02000060 = (*(volatile unsigned int *)0x02000060 & ~0xF) | 1;
+		*(volatile unsigned int *)0x02000070 |= 1;
+	}
 	preloader_console_init();
 
 #if CONFIG_IS_ENABLED(I2C) && CONFIG_IS_ENABLED(SYS_I2C_LEGACY)
