Index: u-boot-src/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-src.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-src/arch/arm/mach-sunxi/board.c
@@ -469,6 +469,10 @@ void board_init_f(ulong dummy)
 	gpio_init();
 
 	spl_init();
+
+	/* --- STATUS LED PC0 ON --- */
+	*(volatile unsigned int *)(0x02000060) = (*(volatile unsigned int *)(0x02000060) & 0xFFFFFFF0) | 0x00000001;
+	*(volatile unsigned int *)(0x02000070) |= 0x00000001;
 	preloader_console_init();
 
 #if CONFIG_IS_ENABLED(I2C) && CONFIG_IS_ENABLED(SYS_I2C_LEGACY)
