Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -476,6 +476,13 @@ void board_init_f(ulong dummy)
 	i2c_init_board();
 	i2c_init(CONFIG_SYS_I2C_SPEED, CONFIG_SYS_I2C_SLAVE);
 #endif
+	/* --- HW INIT (UART/LED) --- */
+	*(volatile unsigned int *)(0x0200190C) |= 0x00010001;
+	*(volatile unsigned int *)(0x02000128) = (*(volatile unsigned int *)(0x02000128) & 0xFFFFF00F) | 0x00000770;
+	*(volatile unsigned int *)(0x02000060) = (*(volatile unsigned int *)(0x02000060) & 0xFFFFFFF0) | 0x00000001;
+	*(volatile unsigned int *)(0x02000070) |= 0x00000001;
+	puts("[DEBUG] HW Init Done in board_init_f\n");
+
 	sunxi_board_init();
 }
 #endif /* CONFIG_XPL_BUILD */
