--- a/arch/arm/mach-sunxi/board.c
+++ b/arch/arm/mach-sunxi/board.c
@@ -10,6 +10,15 @@ void board_init_f(ulong dummy)
	spl_init();
+
+	/* DEBUG LED PC0: ON -> DELAY -> OFF */
+	*(volatile unsigned int *)(0x02000060) &= ~(0xf << 0);
+	*(volatile unsigned int *)(0x02000060) |= (1 << 0);
+	*(volatile unsigned int *)(0x02000070) |= (1 << 0);
+	for (volatile int i = 0; i < 2000000; i++);
+	*(volatile unsigned int *)(0x02000070) &= ~(1 << 0);
+	for (volatile int i = 0; i < 2000000; i++);
+
	preloader_console_init();
+
+	/* AFTER SERIAL: LED ON AGAIN */
+	*(volatile unsigned int *)(0x02000070) |= (1 << 0);
+
}
