--- a/arch/arm/mach-sunxi/board.c
+++ b/arch/arm/mach-sunxi/board.c
@@ -175,1 +175,14 @@
 	spl_init();
+
+	/* UART0 PG17/18 Pinmux Force (Function 7) */
+	*(volatile unsigned int *)(0x020000D8) = (*(volatile unsigned int *)(0x020000D8) & 0xFFFF00FF) | 0x00007700;
+	/* LED PC0 Config (Output) */
+	*(volatile unsigned int *)(0x02000060) = (*(volatile unsigned int *)(0x02000060) & 0xFFFFFFF0) | 0x00000001;
+	/* LED PC0 ON */
+	*(volatile unsigned int *)(0x02000070) |= 0x00000001;
+
+	/* SPL HEARTBEAT: BLINK LED */
+	for (volatile int i = 0; i < 2000000; i++);
+	*(volatile unsigned int *)(0x02000070) &= ~0x00000001; /* OFF */
+	for (volatile int i = 0; i < 2000000; i++);
+	*(volatile unsigned int *)(0x02000070) |= 0x00000001;  /* ON */
