--- a/arch/arm/mach-sunxi/board.c
+++ b/arch/arm/mach-sunxi/board.c
@@ -342,0 +343,11 @@
+	*(volatile unsigned int *)(0x02000060) &= ~(0xf << 0);
+	*(volatile unsigned int *)(0x02000060) |= (1 << 0);
+	*(volatile unsigned int *)(0x02000070) |= (1 << 0);
+	{
+		volatile int i;
+		for(i=0; i<2000000; i++);
+		*(volatile unsigned int *)(0x02000070) &= ~(1 << 0);
+		for(i=0; i<2000000; i++);
+		*(volatile unsigned int *)(0x02000070) |= (1 << 0);
+		for(i=0; i<2000000; i++);
+	}
