Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -476,6 +476,9 @@ void board_init_f(ulong dummy)
 	i2c_init_board();
 	i2c_init(CONFIG_SYS_I2C_SPEED, CONFIG_SYS_I2C_SLAVE);
 #endif
+	/* --- HW INIT --- */
+	puts("[DEBUG] Board Init (UART3 Fix) Reached\n");
+
 	sunxi_board_init();
 }
 #endif /* CONFIG_XPL_BUILD */
