--- a/arch/arm/mach-sunxi/board.c
+++ b/arch/arm/mach-sunxi/board.c
@@ -337,9 +337,20 @@
 
 void board_init_f(ulong dummy)
 {
 	spl_init();
 
+	/* DEBUG LED PC0: ON -> DELAY -> OFF */
+	*(volatile unsigned int *)(0x02000060) &= ~(0xf << 0);
+	*(volatile unsigned int *)(0x02000060) |= (1 << 0);
+	*(volatile unsigned int *)(0x02000070) |= (1 << 0);
+	for(volatile int i=0; i<2000000; i++); 
+	*(volatile unsigned int *)(0x02000070) &= ~(1 << 0);
+	for(volatile int i=0; i<2000000; i++);
+
 	preloader_console_init();
 
+	/* AFTER SERIAL: LED ON AGAIN */
+	*(volatile unsigned int *)(0x02000070) |= (1 << 0);
+
 #if defined(CONFIG_SPL_I2C) && defined(CONFIG_SPL_SYS_I2C_LEGACY)
 	/* This can cost up to 20ms of time, but we may need it for DRAM */
