Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -468,8 +468,22 @@ void board_init_f(ulong dummy)
 	timer_init();
 	gpio_init();
 
+	{ /* Stage K: Physical Setup */
+		*(volatile unsigned int *)0x0200190C = 0x00010001;
+		*(volatile unsigned int *)0x02001128 |= (1 << 0);
+		*(volatile unsigned int *)0x02000128 = 0x77777777; /* PG_CFG2 (PG17/18 UART0) */
+		*(volatile unsigned int *)0x02000060 = 0x11111111; /* PC_CFG0 (PC0 OUT) */
+		*(volatile unsigned int *)0x02000070 |= (1 << 0);
+		*(volatile unsigned int *)0x02000124 = 0x11111111; /* PG_CFG1 (PG11 OUT) */
+		*(volatile unsigned int *)0x02000130 |= (1 << 11);
+		volatile unsigned int *u0 = (volatile unsigned int *)0x02500000;
+		u0[3] = 0x83; u0[0] = 0x0D; u0[1] = 0x00; u0[3] = 0x03;
+		u0[0] = 0x4B; /* SEND K */
+	}
 	spl_init();
+	((volatile unsigned int *)0x02500000)[0] = 0x44; /* SEND D */
 	preloader_console_init();
+	((volatile unsigned int *)0x02500000)[0] = 0x4D; /* SEND M */
 
 #if CONFIG_IS_ENABLED(I2C) && CONFIG_IS_ENABLED(SYS_I2C_LEGACY)
 	/* Needed early by sunxi_board_init if PMU is enabled */
