Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -459,6 +459,23 @@ u32 spl_mmc_boot_mode(struct mmc *mmc, c
 
 void board_init_f(ulong dummy)
 {
+	{ /* T113-i Physical Force Unlock */
+		*(volatile unsigned int *)0x0200190C = 0x00010001; /* PIO Gate */
+		*(volatile unsigned int *)0x0200184C = 0x00010001; /* MMC0 Gate */
+		*(volatile unsigned int *)0x02001830 = 0x80000000; /* MMC0 Module Clock */
+		*(volatile unsigned int *)0x02001128 |= (1 << 0);  /* UART0 Gate */
+		*(volatile unsigned int *)0x020000F0 = 0x00222222; /* MMC Pins */
+		*(volatile unsigned int *)0x02000100 = 0x00333333;
+		*(volatile unsigned int *)0x02000104 = 0x00000555;
+		*(volatile unsigned int *)0x02000128 = 0x77777777; /* UART0 PG Path + Func */
+		*(volatile unsigned int *)0x02000124 = (*(volatile unsigned int *)0x02000124 & ~(0xF << 12)) | (0x1 << 12);
+		*(volatile unsigned int *)0x02000130 |= (1 << 11);
+		*(volatile unsigned int *)0x02000060 = 0x00000001; *(volatile unsigned int *)0x02000070 |= 1;
+		volatile unsigned int *u0 = (volatile unsigned int *)0x02500000;
+		#ifdef CONFIG_SPL_BUILD
+			u0[3] = 0x83; u0[0] = 0x0D; u0[1] = 0x00; u0[3] = 0x03; u0[0] = 0x4B; /* K */
+		#endif
+	}
 	sunxi_sram_init();
 
 	/* Enable non-secure access to some peripherals */
@@ -520,3 +537,6 @@ void enable_caches(void)
 	dcache_enable();
 }
 #endif
+
+#include <spl.h>
+void board_boot_order(u32 *spl_boot_list) { spl_boot_list[0] = BOOT_DEVICE_MMC1; }
