Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c
@@ -469,6 +469,16 @@ void board_init_f(ulong dummy)
 	gpio_init();
 
 	spl_init();
+	/* --- T113-i REPLICA HACK --- */
+	{
+		/* 强制开启 PIO 和 MMC0 的时钟大门，防止总线卡死 */
+		*(volatile unsigned int *)0x0200190C = 0x00010001;
+		*(volatile unsigned int *)0x0200184C = 0x00010001;
+
+		/* 保持 LED PC0 点亮作为生命迹象 */
+		*(volatile unsigned int *)0x02000060 = (*(volatile unsigned int *)0x02000060 & ~0xF) | 1;
+		*(volatile unsigned int *)0x02000070 |= 1;
+	}
 	preloader_console_init();
 
 #if CONFIG_IS_ENABLED(I2C) && CONFIG_IS_ENABLED(SYS_I2C_LEGACY)
