--- a/arch/arm/mach-sunxi/board.c
+++ b/arch/arm/mach-sunxi/board.c
@@ -179,6 +179,15 @@ void board_init_f(ulong dummy)
 	spl_init();
 	preloader_console_init();
 
+	/* DEBUG LED PC0: ON -> DELAY -> OFF */
+	*(volatile unsigned int *)(0x02000060) &= ~(0xf << 0);
+	*(volatile unsigned int *)(0x02000060) |= (1 << 0);
+	*(volatile unsigned int *)(0x02000070) |= (1 << 0);
+	for (volatile int i = 0; i < 2000000; i++);
+	*(volatile unsigned int *)(0x02000070) &= ~(1 << 0);
+	for (volatile int i = 0; i < 2000000; i++);
+
 	/* AFTER SERIAL: LED ON AGAIN */
 	*(volatile unsigned int *)(0x02000070) |= (1 << 0);
 
 #if CONFIG_IS_ENABLED(I2C) && CONFIG_IS_ENABLED(SYS_I2C_LEGACY)
 	/* Needed early by sunxi_board_init if PMU is enabled */
