Index: u-boot-src/common/spl/spl_mmc.c
===================================================================
--- u-boot-src.orig/common/spl/spl_mmc.c
+++ u-boot-src/common/spl/spl_mmc.c
@@ -363,6 +363,13 @@ int spl_mmc_load(struct spl_image_info *
 		if (ret) {
 			mmc = NULL;
 			printf("spl: mmc init failed with error: %d\n", ret);
+
+	/* --- DIAG START --- */
+	{ static char b[512] __aligned(64);
+	  if (blk_dread(mmc_get_blk_desc(mmc), 16, 1, b)) {
+	    printf("DIAG: Sec16 Magic: [%.4s]\n", &b[4]);
+	  } }
+	/* --- DIAG END --- */
 			return ret;
 		}
 	}
