Index: u-boot-2025.01/common/spl/spl_mmc.c
===================================================================
--- u-boot-2025.01.orig/common/spl/spl_mmc.c
+++ u-boot-2025.01/common/spl/spl_mmc.c
@@ -54,6 +54,10 @@ int mmc_load_image_raw_sector(struct spl
 		return ret;
 	}
 
+
+	if (spl_image && spl_image->load_addr) {
+		printf("[MMC-DBG] Data=%08x\n", *(unsigned int *)spl_image->load_addr);
+	}
 	return 0;
 }
 
