Index: u-boot-2025.01/common/spl/spl_mmc.c
===================================================================
--- u-boot-2025.01.orig/common/spl/spl_mmc.c
+++ u-boot-2025.01/common/spl/spl_mmc.c
@@ -54,6 +54,13 @@ int mmc_load_image_raw_sector(struct spl
 		return ret;
 	}
 
+
+	if (spl_image->load_addr) {
+		unsigned int *p = (unsigned int *)spl_image->load_addr;
+		printf("[MMC] Sector: 0x%lx ", sector);
+		printf("Data: 0x%08x 0x%08x
+", p[0], p[1]);
+	}
 	return 0;
 }
 
