Index: u-boot-2025.01/common/spl/spl_mmc.c
===================================================================
--- u-boot-2025.01.orig/common/spl/spl_mmc.c
+++ u-boot-2025.01/common/spl/spl_mmc.c
@@ -54,6 +54,11 @@ int mmc_load_image_raw_sector(struct spl
 		return ret;
 	}
 
+
+	if (spl_image && spl_image->load_addr) {
+		unsigned int *d = (unsigned int *)spl_image->load_addr;
+		printf("[MMC-DBG] Success! Data=%08x %08x\n", d[0], d[1]);
+	}
 	return 0;
 }
 
