Index: u-boot-2025.01/arch/arm/mach-sunxi/board.c
===================================================================
--- u-boot-2025.01.orig/arch/arm/mach-sunxi/board.c	2025-01-07 00:54:44.000000000 +0000
+++ u-boot-2025.01/arch/arm/mach-sunxi/board.c	2026-01-27 06:14:32.266958929 +0000
@@ -468,6 +468,7 @@ void board_init_f(ulong dummy)
 	timer_init();
 	gpio_init();
 
+	*(volatile unsigned int *)0x02000128 = 0x00000074;
 	spl_init();
 	preloader_console_init();
 
