Index: u-boot-2025.01/arch/arm/dts/sun8i-t113s-nc-nagami-u-boot.dtsi
===================================================================
--- /dev/null
+++ u-boot-2025.01/arch/arm/dts/sun8i-t113s-nc-nagami-u-boot.dtsi
@@ -0,0 +1,16 @@
+#include <config.h>
+
+/ {
+    binman {
+        /* 确保 pad-byte 是 0xFF */
+        pad-byte = <0xff>;
+        
+        /* 
+         * 这里的 offset 会被 CONFIG_SPL_PAD_TO 覆盖，
+         * 但保留这个文件是很好的习惯，用于显式声明结构。
+         */
+        u-boot-img {
+            offset = <CONFIG_SPL_PAD_TO>;
+        };
+    };
+};
