Index: u-boot-2025.01/arch/arm/dts/sun8i-t113s-nc-nagami-u-boot.dtsi
===================================================================
--- /dev/null
+++ u-boot-2025.01/arch/arm/dts/sun8i-t113s-nc-nagami-u-boot.dtsi
@@ -0,0 +1,9 @@
+#include <config.h>
+/ {
+    binman {
+        filename = "u-boot-sunxi-with-spl.bin";
+        pad-byte = <0xff>;
+        blob { filename = "spl/sunxi-spl.bin"; };
+        u-boot-img { offset = <0x22000>; };
+    };
+};
